tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -121,7 +112,10 @@ static PyObject * THPVariable_arange(PyObject* self, PyObject* args, PyObject* k
     if (r.isNone(1)) {
       auto end = r.scalar(0);
       // NOTE: r.scalartype(X) gives the default dtype if r.isNone(X)
-      auto scalarType = r.isNone(2) && allIntegral({end}) ? at::ScalarType::Long : r.scalartype(2);
+      c10::optional<ScalarType> scalarType = c10::nullopt;
+      if (!r.isNone(2)) {
+        scalarType = r.scalartype(2);
```

Comment:
Could we add a `optional<ScalarType> PythonArgs::toScalarTypeOptional(int i)` helper method that does this?

Reply:
This actually changes the semantics - `c10::nullopt` for scalar type is different than the default dtype, which is what is happened for `r.scalartype(2);`.